core/arch/arm/plat-hikey/conf.mk

```diff
@@ -24,6 +24,8 @@ CFG_NUM_THREADS ?= 8
 CFG_CRYPTO_WITH_CE ?= y
 CFG_WITH_STACK_CANARIES ?= y
 
+$(call force,CFG_DT,y)
```

Comment:
Yes... this patch was already intended as a "only use if you have patched a-t-f" patch... but that doesn't make sense for upstream series... I changed it to your scheme with docs and here set CFG_DT in the parent level makefile.